### ISSUE TITLE
[A11y] Advanced Search - Native Widgets - Instructions - Advanced search form

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#1801]
### Operating System

Windows

### Browser

Edge

### Affected page

Advanced Search

### Admin user?

- [X] Yes

### Issue Description

"Instructions" seems to refer generally to anything that explains a control, not explicitly stating complex instructions.

Multiple widgets are visually able to be understood, but not programatically.  From AI:

> Both WCAG 2.0 and 2.1 require a widget's visible label and instructions (if present) to be programmatically determinable. WCAG 2.1 also requires a widget's visible label and instructions (if present) to be included in its accessible name and description.

Problems:

- Mentioned in another ticket, but the meaning of the dropdown for "any" and "all" cannot be understood programatically, and has no visible label.
- The meanings of "Title/Description" and "Creator" are visually obvious, but not programatically.  The label is there, so this might not be a hard failure, but the grouping of "any/all" and how it relates to these fields is really hard to determine with a screen reader.

### Suggested Fix

I've read that you can supply instructions above a form and that counts for context.  The criteria in AI suggests that is not the case, but I'm not sure how you would programatically associate the complexity of the form with an individual field.  I believe AI may be applying AAA standards on this one (it isn't clear if it's AAA or AA).

In any case, making the "Search tips" box connect to the form should be a huge win.  e.g., making the tab panel have an `id` like "search-tips", then adding `aria-describedby="search-tips"` to the form tag.  Then the first time entering the form, the SR user is made aware there are instructions for the form as a whole.  Using `aria-describedby` makes it the last thing read, so users who know about the form can skip it once they have heard the form information to begin with.

**Comment by jechols** [comment]
Update: as mentioned in a headings ticket, making the "Search tips" heading into an actual heading would also have a large impact because then SR users can easily discover the same thing sighted users see at a glance - that there are complex rules for search.

**Comment by jechols** [comment]
This one needs discussion: I'd forgotten that the `aria-describedby` situation makes it difficult for screen reader users if there's a lot of text, because there's no way to pause or restart in the middle of descriptions.  The "search tips" area is entirely too long for the brevity that is normally desired in an aria description.

In theory the use of this attribute would satisfy WCAG, but in practice it may not actually be that useful.  I think (but this should be double-checked) that giving the search tips area a heading, thus allowing SR users to find it more easily, would be the bigger win here.

**Comment by jechols** [comment]
Search tips aren't form instructions, so I don't believe the criteria applies that I was thinking of.  Making "Search Tips" into an actual heading would alert everybody, sighted or not, of the existence of "extra information" about the search without cluttering up the form in any way, so that's my recommendation.

Since that is already in a different ticket, I'm closing this one.
